video/out/w32_common.c

```diff
@@ -1543,6 +1582,10 @@ static int gui_thread_control(struct vo_w32_state *w32, int request, void *arg)
         if (w32->opts->fullscreen != w32->current_fs)
             reinit_window_state(w32);
         return VO_TRUE;
+    case VOCTRL_GET_FULLSCREEN:
+        *(int *)arg = w32->toggle_fs ^ w32->current_fs;
+        reinit_window_state(w32);
```

Comment:
So this can't be right here. It shouldn't change any state in a VOCTRL_GET_FULLSCREEN request. The intention of the request is to retrieve the state. Though due to not yet cleaned up issues, the vo_control code down the call stack might overwrite any values in opts-> and update to the current values the frontend (i.e. playloop, command.c etc.) knows. The purpose of the GET request is to update the values in the frontend, so hitting the `F` key and similar will toggle the fullscreen state correctly.

So I'm not sure if you've considered this when writing the code. This stuff is pretty confusing and not very well-defined after all.